### PR TITLE
Assistant role gets harder the more you play

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
This PR will make the assistant role more challenging for veteran assistant players. Now for every 200 hours of play as an assistant, you'll get a free random bad quirk round start (max three bad quirks, so it caps at 600hours)
You can now choose the quirks beforehand to avoid getting random ones

## Why It's Good For The Game
~~Makes assistants the ultimate badasses! Who can overcome those challenges will be crowned the ultimate veteran assistant (not really)~~
Assistants are the only role that have the full freedom of having no responsabilities, giving nothing back to the round. This balance that
New players will have plenty of time to adjust to the game before the tick starts.

## Changelog
:cl:
balance: assistants are more challenging the more you play them, every 200 h of play will randomly give you one bad quirk (unless you select one beforehand so the system won't give you one, warning the mood changing ones don't count, only the higher values one will work)
/:cl:

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
Don't lie in the title.

**Comment by Pinkufeck** [comment]
How much do you want to nerf freeform roles ?

**Comment by SuperNovaa41** [comment]
Having this as an option in character prefs, similar to hardcore random (which is basically this btw) would actually be cool. However, force nerfing people from playing a job that you don't like is a genuine case of power going straight to your head.

**Comment by wesoda25** [comment]
Bait

**Comment by SuperNovaa41** [comment]
> 
> 
> Bait

hes been talking to maintainers about it for the last 30 minutes i dont think its bait

**Comment by Qustinnus** [comment]
how has power gone to his head? this is a pr you clown, he can't force this change in it literaly has to be approved and you know this. stop stupidposting.

**Comment by Qustinnus** [comment]
how has power gone to his head? this is a pr you clown, he can't force this change in it literaly has to be approved and you know this. stop stupidposting.

**Comment by PolishUser** [comment]
lol

**Comment by SuperNovaa41** [comment]
> 
> 
> how has power gone to his head? this is a pr you clown, he can't force this change in it literaly has to be approved and you know this. stop stupidposting.

cope, seethe, stop being cringe, etc etc

**Comment by SuperNovaa41** [comment]
Make it a character pref I think that would be cool

**Comment by Ghilker** [comment]
Added a cap of three bad quirks

**Comment by Mirino97** [comment]
Make assistants insta-debrain if they leave the station. You wanna explore, dipshit?
Open a private server.

**Comment by ArcaneMusic** [comment]
> 
> 
> Just start them off with no cash as unpaid interns. You should not actively punish people for playing a role.

This would actively destabilize the economy, and make what few parts people USE economy for even less utilized.


On the actual PR, this just feels... spiteful? If the goal is to prevent people from playing assistant because it's directionless, the solution shouldn't be "de-incentivize the job from being picked", because then you have Prs like this to actively handicap someone from picking a department and duty. You would be better off by implementing a better global job system, so that the head of personell could look at a roster of 20 assistants and an empty medbay, and say "alright, I'm going to promote some of you medbay, you're now a bartender, we could use one security officer, etc". 

It's easy to think of assistants as when you get a migrant dwarf influx in dwarf fortress. You'll get 30 dwarves showing up with no marketable skills whatsoever, so you'll throw them into jobs that they'll grow to get better at, like soldiering, doctoring, engineering. What's more, players who start with assistant will be de incentivized by proxy to start with jobs they'd rather do anyway, even if they planned on not contributing to the station, because of the knowledge they can have their job swapped away from assistant.

tl;dr: Another misguided person claiming to have all the answers provides one answer that probably doesn't work but whatever.

**Comment by Burgermancoding** [comment]
Just griefing people who want to play without responsibilities

**Comment by ArcaneDefence** [comment]
If people wanted this experience they'd just play random hardcore.

**Comment by saprasam** [comment]
░░░░▄▄▄▄▀▀▀▀▀▀▀▀▄▄▄▄▄▄
░░░░█░░░░▒▒▒▒▒▒▒▒▒▒▒▒░░▀▀▄
░░░█░░░▒▒▒▒▒▒░░░░░░░░▒▒▒░░█
░░█░░░░░░▄██▀▄▄░░░░░▄▄▄░░░█
░▀▒▄▄▄▒░█▀▀▀▀▄▄█░░░██▄▄█░░░█
█▒█▒▄░▀▄▄▄▀░░░░░░░░█░░░▒▒▒▒▒█
█▒█░█▀▄▄░░░░░█▀░░░░▀▄░░▄▀▀▀▄▒█
░█▀▄░█▄░█▀▄▄░▀░▀▀░▄▄▀░░░░█░░█
░░█░░▀▄▀█▄▄░█▀▀▀▄▄▄▄▀▀█▀██░█
░░░█░░██░░▀█▄▄▄█▄▄█▄████░█
░░░░█░░░▀▀▄░█░░░█░███████░█
░░░░░▀▄░░░▀▀▄▄▄█▄█▄█▄█▄▀░░█
░░░░░░░▀▄▄░▒▒▒▒░░░░░░░░░░█
░░░░░░░░░░▀▀▄▄░▒▒▒▒▒▒▒▒▒▒░█
░░░░░░░░░░░░░░▀▄▄▄▄▄░░░░░█

File: code/controllers/subsystem/processing/quirks.dm [hunk 1]
```diff
@@ -68,6 +68,21 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 	if(cli.get_exp_living(TRUE) < EXP_ASSIGN_WAYFINDER && !user.has_quirk(/datum/quirk/needswayfinder))
 		user.add_quirk(/datum/quirk/needswayfinder, TRUE)
 
+	var/list/player_records = cli.prefs.exp
+	var/assistant_playtime = text2num(player_records["Assistant"])
+	var/bad_quirks_to_add = assistant_playtime % 100
```
Comment (MrStonedOne) [inline code/controllers/subsystem/processing/quirks.dm:73]: Playtime is tracked by minute, not hour, internally.

**Comment by FernandoJ8** [comment]
Why must you do these things.
Why...

**Comment by ATH1909** [comment]
this is incredibly redundant with hardcore mode and punishes people for screwing around/vibing as an assistant instead of choosing another job, abandoning their duties, and THEN screwing around.

File: code/controllers/subsystem/processing/quirks.dm (hunk 1 above)
Comment (Ghilker) [reply code/controllers/subsystem/processing/quirks.dm]: fixed

**Comment by Ghilker** [comment]
i've doubled the time to 200 hours, now every 200 h you'll get a random bad quirk

**Comment by Ziiro** [comment]
If your solution can be bypassed in a way that fucks over other players (I don't want brain damage, so I'll just steal X role) rather than you're intended target, it's not a solution.

Try again

**Comment by EdgeLordExe** [comment]
Ghilker, i talked to you multiple times, when i was in tg cord at least. You always were a cool person to talk to, but GOD DAMN, your prs and some contributions are hellish. There is no point in making the game arbitrarily more tedious, balancing via tedium is something i did with my explosive chems, and it caused chemists life not more fun, not more difficult, but more boring. I understand that you like to balance things, and getting shit merged is a big dopamine hit, but please fucking reconsider your stance on balance.

**Comment by Time-Green** [comment]
> i've doubled the time to 200 hours, now every 200 h you'll get a random bad quirk

Don’t give in to the haters. A bad quirk for every hour played seems more fair, since then the basic controles have been learned and they can start focusing on real jobs

**Comment by tommysalami3** [comment]
i want this change because it lets me flex my assistant playtime on non-mentally-damaged individuals. speedmerge.

**Comment by Farquaar** [comment]
Quality bait. Almost got me to bite, and assistant is probably my least-played role.

**Comment by StarHorder** [comment]
Who was the one that gave me dozens of strange objects? an assisstant.
who was the one that made maintbar look pretty? an assisstant.
who was the one that singlehandedly prevented a crystal invasion? an assisstant.

**Comment by tommysalami3** [comment]
> Who was the one that gave me dozens of strange objects? an assisstant.
> who was the one that made maintbar look pretty? an assisstant.
> who was the one that singlehandedly prevented a crystal invasion? an assisstant.

thats cool but can you change your profile picture to actually anything else

**Comment by maxymax13** [comment]
> 
> 
> Who was the one that gave me dozens of strange objects? an assisstant.
> who was the one that made maintbar look pretty? an assisstant.
> who was the one that singlehandedly prevented a crystal invasion? an assisstant.

spell assistant right

**Comment by maxymax13** [comment]
ghilker vs mso final boss battle lets go

**Comment by optimumtact** [comment]
Can we also give any assistant with more than 30 hours play time ticking brain damage for every second they are alive in a game?

This will represent their real life mental degradation and be funny.

**Comment by Ghilker** [comment]
> If your solution can be bypassed in a way that fucks over other players (I don't want brain damage, so I'll just steal X role) rather than your intended target, it's not a solution.
> 
> Try again

That's the intention, moving people to play other jobs instead of the new player one. You are not bypassing the system, you are doing what the system intended.

**Comment by RaveRadbury** [comment]
Consider letting them pick the bad quirks, only assigning them at random if they don't have enough set.

**Comment by cacogen** [comment]
This is fucking stupid

**Comment by maxymax13** [comment]
> 
> 
> > If your solution can be bypassed in a way that fucks over other players (I don't want brain damage, so I'll just steal X role) rather than your intended target, it's not a solution.
> > Try again
> 
> That's the intention, moving people to play other jobs instead of the new player one. You are not bypassing the system, you are doing what the system intended.

so are you saying your system intends for tiders to take all the engi and cargo job slots to do even less than the average engineer/cargonian

**Comment by cacogen** [comment]
Yeah, if people don't want to do a job forcing them to take a job slot isn't going to change that

**Comment by Naloacisamazing** [comment]
wow this is one late april fools pr 🤡

**Comment by CyoTheProtogen** [comment]
Time to play station engineer instead and continue to tide

**Comment by MrStonedOne** [comment]
smurf accounts, /tg/ edition!

**Comment by SuperNovaa41** [comment]
You should apply this to every role so we can finally kill off this game

**Comment by Ghilker** [comment]
> Consider letting them pick the bad quirks, only assigning them at random if they don't have enough set.

done, now if you choose some of the bad quirks (the ones with the higher negative values, so not the mood ones) you won't get them randomly chosen. so if you have 200 h of play, and you select bad aim, you'll only have bad aim on you and no other random quirks, this allows you to also choose good quirks that was previously impossible to do

**Comment by interestingusernam3** [comment]
lol

**Comment by ChocolateBickie** [comment]
Why don't you focus on making other jobs better/more fun instead of making players lives worse because they don't play the game how you want?

That is the issue here. People play assistant because it's fun and avoid boring or repetitive jobs.

**Comment by ZeroMODE0** [comment]
Allow me to join the game late, have the HOP close all jobs but assistant as a meme, (or let it be highpop and all jobs are taken) and be forced to join as assistant with bad quirks or to observe instead. Alternatively, my preferred jobs were taken at round start and I am now a disabled assistant instead of my preferred jobs. As an added bonus, if I'm given certain bad quirks like bad aim or narcolepsy I can't even help the station by mining when miners are dead.
Can we just not do this? Please?

**Comment by Mirino97** [comment]
I now understand why Oranges is such a prick.
Bait is fun!

**Comment by Mothblocks** [comment]
funny but i don't like it

**Comment by Ghilker** [comment]
He didn't get away with it this time
STolLU9B
